janitor/functions.py

```diff
@@ -3150,7 +3150,19 @@ def update_where(
     :raises: IndexError if **conditions** does not have the same length as
         **df**.
     """
-    df.loc[conditions, target_column_name] = target_val
+    # use query style method
+    # to accommodate changes if the dataframe has been modified
+    # somewhere along the method chain block
+    # suggested by @zbarry
+
+    if not isinstance(conditions, str):
+        raise TypeError("conditions argument should be an expression")
```

Comment:
A hopefully more helpful error message.

```suggestion
    if not isinstance(conditions, str):
        raise TypeError("conditions argument should be a pandas-compatible string query. See https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html for more information.")
```